From 4aedccab587a514c951b1a23dfd87bbcd0e42ffd Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Sun, 27 Jul 2014 20:44:40 -0400
Subject: [net] sctp: only warn in proc_sctp_do_alpha_beta if write

Message-id: <0f12de2a95392965a78d09c06f9300557007fd9a.1406493713.git.dborkman@redhat.com>
Patchwork-id: 86723
O-Subject: [RHEL7 PATCH net 8/7] net: sctp: only warn in proc_sctp_do_alpha_beta if write
Bugzilla: 1110290
RH-Acked-by: Flavio Leitner <fbl@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110290
Upstream Status: eaea2da7286ebc56d557b40ad7e59e715a84e4a0 (net-next)
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7738424
Tested: verified on beaker machine

Upstream Commit:
commit eaea2da7286ebc56d557b40ad7e59e715a84e4a0
Author: Daniel Borkmann <dborkman@redhat.com>
Date:   Mon Jun 30 13:52:09 2014 +0200

    net: sctp: only warn in proc_sctp_do_alpha_beta if write

    Only warn if the value is written to alpha or beta. We don't care
    emitting a one-time warning when only reading it.

    Reported-by: Jiri Pirko <jpirko@redhat.com>
    Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
    Reviewed-by: Jiri Pirko <jiri@resnulli.us>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sctp/sysctl.c b/net/sctp/sysctl.c
index e81312b..333973c 100644
--- a/net/sctp/sysctl.c
+++ b/net/sctp/sysctl.c
@@ -431,8 +431,9 @@ static int proc_sctp_do_alpha_beta(struct ctl_table *ctl, int write,
        void __user *buffer, size_t *lenp,
        loff_t *ppos)
 {
- pr_warn_once("Changing rto_alpha or rto_beta may lead to "
-       "suboptimal rtt/srtt estimations!\n");
+ if (write)
+  pr_warn_once("Changing rto_alpha or rto_beta may lead to "
+        "suboptimal rtt/srtt estimations!\n");
 
  return proc_dointvec_minmax(ctl, write, buffer, lenp, ppos);
 }
-- 
1.7.1